common/src/main/java/com/wynntils/features/user/TooltipScaleFeature.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.properties.ConfigOption;
+import com.wynntils.core.config.properties.Configurable;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.EventListener;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.FeatureInfo.Stability;
+import com.wynntils.mc.event.ItemTooltipRenderEvent;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.utils.WynnUtils;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@EventListener
+@Configurable(category = "Item Tooltips")
+@FeatureInfo(stability = Stability.STABLE)
+public class TooltipScaleFeature extends UserFeature {
+
+    @ConfigOption(
+            displayName = "Universal Scale",
+            description = "The scale factor that should be applied to every tooltip")
+    public static float universalScale = 1f;
+
+    @ConfigOption(
+            displayName = "Fit to Screen",
+            description = "Whether tooltips should be scaled to always fit on the screen")
+    public static boolean fitToScreen = true;
+
+    private boolean scaledLast = false;
+    private Screen currentScreen = null;
+    private int oldWidth = -1;
+    private int oldHeight = -1;
+
+    @SubscribeEvent
+    public void onTooltipPre(ItemTooltipRenderEvent.Pre e) {
+        if (!WynnUtils.onServer()) return;
+
+        currentScreen = McUtils.mc().screen;
+        if (currentScreen == null) return; // shouldn't be possible
+
+        // calculate scale factor
+        float scaleFactor = universalScale;
+
+        if (fitToScreen) {
+            int lines = ItemUtils.getTooltipLines(e.getItemStack()).size();
+            // this is technically slightly larger than the actual height, but due to the tooltip offset/border, it
+            // works to create a nice buffer at the top/bottom of the screen
+            float tooltipHeight = 22 + (lines - 1) * 10;
+            tooltipHeight *= universalScale;
+
+            if (tooltipHeight > currentScreen.height) scaleFactor *= (currentScreen.height / tooltipHeight);
```

Comment:
it is effectively cancelled, yes, but it needs to be included here to account for the possibility of universalScale being larger than 1, and causing a tooltip to go off the screen when it otherwise wouldn't

Reply:
the features are not mutually exclusive, which is something i was careful to make sure of

Reply:
If its being cancelled out then fittoscreen effectively overrides universalscale

Reply:
yes - _only_ in the case that the scaled up tooltip would be larger than the screen. otherwise the tooltip is scaled up and nothing else happens

Reply:
Oh yeah im crazy.